btx/interfaces/psana_interface.py

```diff
@@ -213,19 +244,20 @@ def get_images(self, num_images, assemble=True):
                 
             else:
                 evt = self.runner.event(self.times[self.counter])
-                if assemble:
-                    if not self.calibrate:
-                        raise IOError("Error: calibration data not found for this run.")
-                    else:
-                        images[counter_batch] = self.det.image(evt=evt)
-                else:
-                    if self.calibrate:
-                        images[counter_batch] = self.det.calib(evt=evt)
+                if not self.skip_event(evt):
```

Comment:
It seems like this will no longer retrieve `num_images` events; maybe the logic needs to be changed to a while loop instead?

Reply:
I removed this. See https://github.com/lcls-users/btx/pull/106/commits/d1d817b41c651e715241b5928f91f91ec59e0171